openai-engine.el

```diff
@@ -38,15 +38,16 @@
                                &key
                                (content-type "application/json")
                                (key openai-key)
-                               org-id)
+                               org-id
+                               (base-url openai-base-url))
```

Comment:
I have the preference to place variables in order depends on when the variable is in used. In this case, `base-url` comes to the very top since that's the variable we see on the first line of our function.

```elisp
(openai-request (concat base-url "/engines")   ; <- `base-url` go first
    :type "GET"
    :headers (openai--headers content-type key org-id)
    :parser 'json-read
```

Reply:
Ah, okay. I mean "very top" but actually under `&key`, so it should be something like:

```diff
(cl-defun openai-engine-list ( callback
                               &key
+                               (base-url openai-base-url) 
                               (content-type "application/json")
                               (key openai-key)
                               org-id)
```

The order of the arguments followed this rule:

```
(cl-defun openai-xxx ( [required] &key [optional])
```

In this example, `callback` is required or else the function wouldn't work or being meaningless. If we don't pass the argument `callback` in, then the function will either give you error 400 or be meaningless.

Reply:
Done